### PR TITLE
Add a --version switch for adding file version keys

### PR DESCRIPTION
**Pull request by rillian**
Add a command-line switch for inserting the `version` keys we've been using to mark toolchain records in gecko manifests.

### CODE REVIEW

File: test_tooltool.py [hunk 1]
```diff
@@ -315,7 +315,16 @@ def test_json_dumps_with_unpack_and_setup(self):
             self.test_record, cls=tooltool.FileRecordJSONEncoder)
         from_json = json.loads(json_string,
                                cls=tooltool.FileRecordJSONDecoder)
-        for i in ['filename', 'size', 'algorithm', 'digest', 'unpack', 'setup']:
+        for i in ['filename', 'size', 'algorithm', 'digest','unpack', 'setup']:
```
Comment (djmitche) [inline test_tooltool.py]: dropped a space here

Comment (rillian) [inline test_tooltool.py]: Thanks. Fixed.

**Comment by rillian** [comment]
> Is this going to produce manifests that the old version can't parse, or be unable to parse old manifests?

I think it will be ok. It doesn't output a `version` key if none is set so old scripts should see the same behaviour, and we've had extra keys in the in-tree manifests for some time without incident so I thought tooltool ignored them.

This just adds official support for a manifest key we were inserting manually.

**Comment by djmitche** [comment]
Sounds good!

Do you mind making a bug/patch to update the Gecko in-tree version, too?

**Comment by rillian** [comment]
Sure. I think it would be helpful to have some tags tracking useful adoption points. I filed #37 in support of this idea. :)

**Pull request merged**